src/api/procedures/investInSto.ts

```diff
@@ -31,10 +31,53 @@ export type Params = InvestInStoParams & {
  * @hidden
  */
 export interface Storage {
-  investmentPortfolioId: PortfolioId;
+  purchasePortfolioId: PortfolioId;
   fundingPortfolioId: PortfolioId;
 }
 
+/**
+ * @hidden
+ */
+type TierStats = {
+  remaining: BigNumber;
+  price: BigNumber;
+  remainingAmount: BigNumber;
+};
+
+/**
+ * @hidden
+ */
+export const calculateTierStats = (
+  tiers: Tier[],
+  purchaseAmount: BigNumber,
+  maxPrice?: BigNumber
+): TierStats => {
+  return tiers.reduce<TierStats>(
+    (prev, { remaining, price }) => {
+      if ((!maxPrice || price.lte(maxPrice)) && !prev.remainingAmount.isZero()) {
+        if (remaining.gte(prev.remainingAmount)) {
+          return {
+            remaining: prev.remaining.plus(prev.remainingAmount),
+            price: prev.price.plus(prev.remainingAmount.multipliedBy(price)),
+            remainingAmount: new BigNumber(0),
+          };
+        }
+        return {
+          remaining: prev.remaining.plus(remaining),
+          price: prev.price.plus(remaining.multipliedBy(price)),
+          remainingAmount: prev.remainingAmount.minus(remaining),
+        };
+      }
+      return prev;
+    },
+    {
+      remaining: new BigNumber(0),
+      price: new BigNumber(0),
+      remainingAmount: purchaseAmount,
```

Comment:
`remainingToPurchase`

Also, this shouldn't be returned from the function, it's just an internal counter. You should just destructure the other two properties from the result of the `reduce` and return them, leaving this one out